src/lib/fragments/AgGrid.react.js

```diff
@@ -192,15 +194,66 @@ export default class DashAgGrid extends Component {
 
     setSelection(selection) {
         const {gridApi} = this.state;
+        const {getRowId} = this.props;
         if (gridApi && selection) {
-            if (!selection.length) {
+            this.setState({pauseSelections: true});
+            if (Object.keys(selection).includes('function')) {
+                // keeps grid from rendering display unnecessarily
                 gridApi.deselectAll();
-            } else {
+                const parsedCondition = esprima.parse(selection.function)
+                    .body[0].expression;
+                const context = {
+                    d3,
+                    ...customFunctions,
+                    ...window.dashAgGridFunctions,
+                };
+                const test = function (node) {
+                    if (evaluate(parsedCondition, {node, ...context})) {
+                        node.setSelected(true);
+                    }
+                };
+                gridApi.forEachNode(test);
```

Comment:
Does it not work to use `this.parseFunction` here?
```suggestion
                const test = this.parseFunction(selection.function);
                gridApi.forEachNode((node) => {
                    if (test(node)) {
                        node.setSelected(true);
                    }
                });
```

Reply:
OIC `parseFunction` puts its argument into `params` and you want it in `node`. That name could be added as an optional argument to `parseFunction` so we can still reuse the rest of that.
```js
parseFunction = memoizeWith(String, (funcString, argName = 'params') => {
...
    return (params) => evaluate(parsedCondition, {[argName]: params, ...context});
```
then you'd call it with:
```js
const test = this.parseFunction(selection.function, 'node');
```

Reply:
I think we could just leave it as `params`, that way there is unity among all of our functions. XD

Reply:
Haha yeah actually, seems like `params` is more consistent with other functions anyway, so https://github.com/plotly/dash-ag-grid/pull/164/commits/10b361c3b00ab0383b88d565da8e8273f9658207 is good 🎉